### PR TITLE
fix pcl version 1.10 compatibility with ndt

### PR DESCRIPTION
**Pull request by msmcconnell**
# PR Details
## Description
This PR resolves https://github.com/usdot-fhwa-stol/carma-platform/issues/1559 by updating the NDT optimization parameters to reflect the new PCL 1.9+ API changes.

ROS Noetic targets PCL v 1.10 whereas ROS Kinetic target PCL v 1.7.2
It seems that in v1.9 the API for the error epsilons was changed (see https://github.com/PointCloudLibrary/pcl/pull/1724) Now the translation epsilon is actually a squared error and there is an additional epsilon which is the rotation angle error in an axis-angle rotation representation. This PR updates the translation error setting in NDT to therefore be 0.001 which makes the optimization selectivity closer to the old value (TODO I'm still evaluating if 0.001 or 0.0001 (which is 0.01*0.01) is better) (No longer investigating, see comment below). The rotation error is computed as 1.0-translation error which is the default used technique in the ICP matching algorithm (seems they didn't add a default for NDT matching). This may seem strange but actually makes sense as the input is cos(angle error). And arccos(0.999) =0.044 rad or 2.5 deg error.  

This PR also removes the --sequential build argument from the docker image to hopefully speed up build times. 

## Related Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/1559

## Motivation and Context
Stable localization in Elise

## How Has This Been Tested?
Initial testing conducted on Blue lexus shows good results. 

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by msmcconnell** [comment]
I performed some comparison runs using translation epsilons of 0.001 and 0.0001 respectively. Both gave similar over all NDT score values. The first value required fewer iterations overall and hit the max less often while the second value more often hit the max. Both ran within reasonable time frames though the second value did go above 100ms more frequently. 

Since I did not see a noticeable improvement in NDT score, I plan to continue with 0.001 for now as it stayed under the 100ms execution time target more reliably. However, I have left a comment describing that 0.0001 can work as documentation of its feasibility.

File: core_perception/lidar_localizer/nodes/ndt_matching/ndt_matching.cpp [hunk 1]
```diff
@@ -130,8 +130,9 @@ static pcl_omp::NormalDistributionsTransform<pcl::PointXYZ, pcl::PointXYZ> omp_n
 static int max_iter = 30;        // Maximum iterations
 static float ndt_res = 1.0;      // Resolution
 static double step_size = 0.1;   // Step size
-static double trans_eps = 0.01;  // Transformation epsilon
-
+static double trans_eps = 0.001;  // Transformation epsilon. In PCLv1.10 (ros noetic) this value is squared error not base epsilon 
+                                  // NOTE: A value of 0.0001 can work as well. 
+                                  // This will increase the required iteration count (and therefore execution time) but might increase performance
```
Comment (kjrush) [inline core_perception/lidar_localizer/nodes/ndt_matching/ndt_matching.cpp]: Just to clarify, the mentioned performance here is in terms of accuracy? So the tradeoff is speed vs accuracy?

File: core_perception/lidar_localizer/nodes/ndt_matching/ndt_matching.cpp [hunk 2]
```diff
@@ -322,18 +323,27 @@ static void param_callback(const autoware_config_msgs::ConfigNDT::ConstPtr& inpu
   if (input->trans_epsilon != trans_eps)
   {
     trans_eps = input->trans_epsilon;
+    double rot_threshold = 1.0 - trans_eps;
+    
 
-    if (_method_type == MethodType::PCL_GENERIC)
+    if (_method_type == MethodType::PCL_GENERIC) {
+      ROS_INFO_STREAM("Using tramslation threshold of " << trans_eps);
```
Comment (kjrush) [inline core_perception/lidar_localizer/nodes/ndt_matching/ndt_matching.cpp]: Typo? `tramslation`

**Review by kjrush — APPROVED** [review]

File: core_perception/lidar_localizer/nodes/ndt_matching/ndt_matching.cpp (hunk 1 above)
Comment (msmcconnell) [reply core_perception/lidar_localizer/nodes/ndt_matching/ndt_matching.cpp]: Yeah, I'll clarify

File: core_perception/lidar_localizer/nodes/ndt_matching/ndt_matching.cpp (hunk 2 above)
Comment (msmcconnell) [reply core_perception/lidar_localizer/nodes/ndt_matching/ndt_matching.cpp]: Fixed

**Pull request merged**